lib/add_route/add_page.dart

```diff
@@ -116,35 +92,43 @@ class _AddPageState extends State<AddPage> {
     }
   }
 
-  void _save() {
+  Future<void> _save() async {
     if (_titleController.text != "") {
       if (_date != null) {
-        DateTime date = _date!;
         if (_selectedType != null) {
-          RouteType type = _selectedType!;
           if (_grade != null) {
-            String grade = _grade!;
             if (_status != null) {
-              Status status = _status!;
-              if ((_status == Status.completed && _completedStatus != null) ||
-                  _status != Status.completed) {
-                if (_status != Status.completed) {
+              if ((_status == RouteStatusEnum.completed &&
+                      _completedStatus != null) ||
+                  _status != RouteStatusEnum.completed) {
+                if (_status != RouteStatusEnum.completed) {
                   _completedStatus = null;
                 }
 
-                ClimbingRoute route = ClimbingRoute(
-                    _titleController.text,
-                    date,
-                    type,
-                    grade,
-                    status,
-                    _completedStatus,
-                    Difficulty.values[_difficultyIndex.round()],
-                    _selectedTags,
-                    _thoughtsController.text);
-                print(route.toString());
+                var db = Provider.of<AppDatabase>(context, listen: false);
+
+                RouteCompanion newRoute = RouteCompanion(
+                  title: drift.Value(_titleController.text),
+                  date: drift.Value(_date!),
+                  climbTypeId: drift.Value(_selectedType!.index),
+                  gradeSystemId: drift.Value(_selectedGradeSystem!.index),
+                  grade: drift.Value(_grade!),
+                  status: drift.Value(_status!.index),
+                  completedStatus: drift.Value(_completedStatus?.index),
+                  difficulty: drift.Value(_difficultyIndex.toInt()),
+                  thoughts: drift.Value(_thoughtsController.text),
+                );
+                var insertedRoute = await db.routeDao.insertReturning(newRoute);
 
-                Navigator.pop(context);
+                for (var tag in _selectedTags) {
+                  db.routeTagDao.insert(RouteTagCompanion(
+                    routeId: drift.Value(insertedRoute.id),
+                    tagId: drift.Value(tag.index)
+                  ));
+                }
+
+                if (!mounted) return;
```

Comment:
Nitpick: always use curly braces with if statement

Reply:
I think this way is more concise and is recommended by the dart linter [here](https://dart-lang.github.io/linter/lints/use_build_context_synchronously.html)